src/components/Avatar/Avatar.tsx

```diff
@@ -21,6 +29,140 @@ export const AVATAR_THEME_SET = [
   styles.grey,
 ];
 
+export const getStatusItemSizeAndPadding = (
+  avatarSize: number
+): [number, number] => {
+  const statusItemSize: number = (avatarSize * 16) / 100;
+  return [statusItemSize, (statusItemSize * 6) / 16];
+};
+
+const AvatarStatusItems: FC<BaseAvatarProps> = React.forwardRef(
+  ({ outline, size, statusItems }) => {
+    const htmlDir: string = useCanvasDirection();
+
+    const getPositionForRtl = (
+      statusItemPos: StatusItemsPosition
+    ): StatusItemsPosition => {
+      switch (statusItemPos) {
+        case StatusItemsPosition.Left:
+          return StatusItemsPosition.Right;
+        case StatusItemsPosition.Right:
+          return StatusItemsPosition.Left;
+        case StatusItemsPosition.TopLeft:
+          return StatusItemsPosition.TopRight;
+        case StatusItemsPosition.TopRight:
+          return StatusItemsPosition.TopLeft;
+        case StatusItemsPosition.BottomLeft:
+          return StatusItemsPosition.BottomRight;
+        case StatusItemsPosition.BottomRight:
+          return StatusItemsPosition.BottomLeft;
+        default:
+          return statusItemPos;
+      }
+    };
+
+    const getStatusItemBasicStyle = (
+      statusItemProps: StatusItemsProps
+    ): React.CSSProperties => {
+      return {
+        position: 'absolute',
+        borderRadius: '50%',
```

Comment:
move this to a css class

Reply:
Same as `wrapperContainerStyle`..., and having all the styling at one place is better than having it split between code and class